shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1749,16 +1800,18 @@ - (void)hoverEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(ios(13.4))
   packet->SetPointerData(/*index=*/0, pointer_data);
 
   [_engine.get() dispatchPointerDataPacket:std::move(packet)];
+  _mouseState.flutter_state_is_added = pointer_data.change != flutter::PointerData::Change::kRemove;
```

Comment:
I'm un-sure about this, it looks fine.

CC @chunhtai do you mind take a look at the logics about pointers in this PR?

Reply:
This seems to assume the engine shell will synthesize add event when the change is not `kRemove`? why not always synthesize the add event in embedding so that you don't need to guess what engine shell may be doing

Reply:
The correct events will be sent to the shell. Nothing needs to be synthesized. This is to synchronize so that in hoverEvent and scrollEvent we don't send the same event twice.